firmware/quadruna/FSM/src/tasks.c

```diff
@@ -0,0 +1,130 @@
+#include "tasks.h"
+#include "main.h"
+#include "cmsis_os.h"
+
+#include "app_globals.h"
+#include "app_heartbeatMonitor.h"
+#include "states/app_driveState.h"
+
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "app_canAlerts.h"
+#include "app_commitInfo.h"
+
+#include "io_jsoncan.h"
+
+#include "hw_bootup.h"
+#include "hw_hardFaultHandler.h"
+#include "hw_watchdog.h"
+#include "hw_watchdogConfig.h"
+
+extern ADC_HandleTypeDef  *hadc1;
+extern CAN_HandleTypeDef  *hcan1;
+extern IWDG_HandleTypeDef *hiwdg;
+extern TIM_HandleTypeDef  *htim12;
+struct StateMachine       *state_machine;
+struct HeartbeatMonitor   *heartbeat_monitor;
+
+void canRxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_RxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+void canTxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_TxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+const CanConfig can_config = {
+    .rx_msg_filter        = NULL,
```

Comment:
This can be set to `io_canRx_filterMessageId`.

This is generated from JSONCAN, so this board only needs to decode CAN messages that this board actually receives (i.e. specified in the RX JSON). 

Since a board doesn't know what message its receiving until after its received it, it needs to receive all messages, but then only decodes the payload if this `io_canRx_filterMessageId` says it needs to.

Can you also change this on the VC? Looks like it was missed.

Reply:
Fixed!